synapse/handlers/federation_event.py

```diff
@@ -953,6 +953,42 @@ async def _process_pulled_event(
             )
             return
 
+        # Check if we've already tried to process this event
+        failed_pull_attempt_info = await self._store.get_event_failed_pull_attempt_info(
+            event.room_id, event_id
+        )
+        # TODO: Should we only do this for `backfilled=true`?
```

Comment:
This function is used for backfill and when another server sends us a PDU. In either case, we're trying to avoid the costly state calculations to see if it's allowed so I think we should always do this check